### PR TITLE
Feature/simple graph

### PR DESCRIPTION
**Pull request by jay-tyler**

### CODE REVIEW

File: test_graph.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+        5: set([10]),
+        10: set([5, 20, 15]),
+        15: set(),
+        20: set([5]),
+        25: set(),
+        30: set()
+    }
+    return g
+
+def test_valid_constructor():
```
Comment (Jakeand3rson) [inline test_graph.py]: Should be two spaces here.

File: test_graph.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    out = graph_filled.edges()
+    expected_edges = set([(5, 10), (10, 5), (10, 20), (10, 15), (20, 5)])
+    assert set(out) == expected_edges
+    assert len(out) == 5
+
+
+def test_host_node_empty(graph_empty):
+    unexpected_nodes = set([0, 2, 7, 13, 27, 33])
+    for node in unexpected_nodes:
+        assert graph_empty.has_node(node) == False
```
Comment (Jakeand3rson) [inline test_graph.py]: Convention is 
```
assert graph_empty.has_node(node) is False
```

File: test_graph.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+
+def test_has_node_filled(graph_filled):
+    expected_nodes = set([5, 10, 15, 20, 25, 30])
+    unexpected_nodes = set([0, 2, 7, 13, 27, 33])
+    for node in expected_nodes:
+        assert graph_filled.has_node(node) == True
```
Comment (Jakeand3rson) [inline test_graph.py]: Same thing. Should be ``` is ```

File: test_graph.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    for node in unexpected_nodes:
+        assert graph_filled.has_node(node) == False
```
Comment (Jakeand3rson) [inline test_graph.py]: ``` is ```

File: test_graph.py (hunk 2 above)
Comment (jonathanstallings) [inline test_graph.py]: Good catch! Thanks.

File: test_graph.py (hunk 3 above)
Comment (jonathanstallings) [inline test_graph.py]: Yup!

File: test_graph.py (hunk 1 above)
Comment (jonathanstallings) [inline test_graph.py]: Thank you!

**Pull request merged**